File: snail-race-bet/typescript/src/domain/FakeSnailRacesProvider.ts
import {Podium, SnailRaces, SnailRacesArena} from "./SnailRacesArena";

export class FakeSnailRacesProvider implements SnailRacesArena {
    private snailRaces: SnailRaces;

    constructor(snailRaces?: SnailRaces) {
        this.snailRaces = snailRaces || new SnailRaces([]);

    }

    async races(): Promise<SnailRaces> {
        return this.snailRaces;
    }

    simulateRaceResult(raceId: number, datetime: number, podium: Podium) {
        this.snailRaces = SnailRaces.withAdditionalResult(this.snailRaces, raceId, datetime, podium);
    }
}

File: snail-race-bet/java/src/main/java/snail/race/kata/domain/FakeSnailRacesProvider.java
package snail.race.kata.domain;


import java.util.ArrayList;

public class FakeSnailRacesProvider implements RaceResultProvider {
    private SnailRaces snailRaces;

    public FakeSnailRacesProvider() {
        this.snailRaces = new SnailRaces(new ArrayList<>());
    }

    public FakeSnailRacesProvider(SnailRaces snailRaces) {
        this.snailRaces = snailRaces;
    }

    @Override
    public SnailRaces races() {
        return snailRaces;
    }

    public void simulateRaceResult(int raceId, long datetime, Podium podium) {
        this.snailRaces = SnailRaces.withAdditionalResult(this.snailRaces, raceId, datetime, podium);
    }
}

